### DISCUSSION TITLE
Determining a Message's ChannelType

### DISCUSSION THREAD

**Discussion by DitchTurtL** [bwmarrin/discordgo#1319] [Q&A] [answered]
I would like to determine if a message is a DM or public channel message.

We get ChannelID with the message object, but I can't find a function or path back to Channel.ChannelType.
Is there a way to do this?

**Comment by Nv7-GitHub** [comment]
Fetch the channel using the Session and get it's type from there

**Reply by DitchTurtL** [reply]
I appreciate your experience. I was able to find the answer after reading your reply. 
"Fetch" doesn't appear anywhere functional in the repo and was confusing in your response. 
I've written a slightly more cohesive answer with proper capitalization and an example for folks that may not be as experienced with digging into library code.

Thanks a lot for the input.

**Reply by Nv7-GitHub** [reply]
Sorry, couldn't access discordgo docs when I wrote the answer and didn't remember the method off the top of my head

**Reply by DitchTurtL** [reply]
No worries, appreciate the reply.

**Comment by DitchTurtL** [comment, marked as answer]
We can call the Channel function on the local session variable in messageCreate(session, message).
That will return a reference to the channel where we can get (int)Type and compare to the ChannelType constant to determine the message type.

st, err := s.Channel(m.ChannelID)
st.Type != discordgo.ChannelTypeDM

**Reply by Vasyanhik** [reply]
> We can call the Channel function on the local session variable in messageCreate(session, message).
> That will return a reference to the channel where we can get (int)Type and compare to the ChannelType constant to determine the message type.
> 
> st, err := s.Channel(m.ChannelID)
> st.Type != discordgo.ChannelTypeDM

**Reply by Vasyanhik** [reply]
> We can call the Channel function on the local session variable in messageCreate(session, message).
> That will return a reference to the channel where we can get (int)Type and compare to the ChannelType constant to determine the message type.
> 
> st, err := s.Channel(m.ChannelID)
> st.Type != discordgo.ChannelTypeDM
